### ISSUE TITLE
Unable to make authorized GCS requests

### ISSUE THREAD

**Issue by mikeas1** [igvteam/igv.js#272]
It appears that requests for GCS paths (gs://) are not setting the Authorization header. Looks like this functionality was [removed a while ago](https://github.com/igvteam/igv.js/commit/e3366eb8300c4e7433d05b994e45ca23a0610c72#diff-9bedefe194a1fcdcb075a0633bba1ce5L89). Was there any particular reason?

One option that would probably be more robust would be to use [gapi](https://developers.google.com/api-client-library/javascript/) for interacting with Google APIs, including the GA4GH Cloud Genomics endpoint. This would allow users to configure authorization separately, meaning this library doesn't need to concern itself with auth for those APIs basically at all, I think. This may end up being a significant refactor, though.

Note that there is a work-around by manually supplying headers in the track config, such as:
```
addTrack({
  sourceType: 'gcs',
  type: 'bam',
  url: 'gs://some_bucket/some_object.bam'
  headers: {
    'Authorization': 'Bearer ' + oauthToken
  },
  name: 'A Bam Track'
})
```

**Comment by jrobinso** [comment]
I'll look into why that was removed, probably inadvertent.  The supported
method for google oAuth is docuemented here, namely you set a global
variable: https://github.com/igvteam/igv.js/wiki/Google-Authentication.

What version of igv.js are you using?  A number of projects are using oAuth
successfully with GA4GH and gs:// urls.

There is another issue, which I opened, to support gapi if its present.
However I don't want to require gapi.

**Comment by mikeas1** [comment]
This is with v 1.0.1. Checking back on v1.0.0 it looks like the code for setting the Authorization token is still present.

**Comment by jrobinso** [comment]
I will look into this, and the indexURL problem tomorrow.

**Comment by jrobinso** [comment]
Fixed in v 1.0.5
